core/bin/external_node/src/config/mod.rs

```diff
@@ -249,6 +250,13 @@ pub struct OptionalENConfig {
     // This is intentionally not a part of `RemoteENConfig` because fetching this info from the main node would defeat
     // its purpose; the consistency checker assumes that the main node may provide false information.
     pub contracts_diamond_proxy_addr: Option<Address>,
+    /// Number of permits in each rate limiting window applied to the main node HTTP client. Default is 10 permits.
+    #[serde(default = "OptionalENConfig::default_main_node_rate_limit_count")]
+    pub main_node_rate_limit_count: NonZeroUsize,
+    /// Duration of a rate limiting window applied to the main node HTTP client. Default is 50 ms.
+    /// Setting this duration to 0 will effectively disable rate limiting.
+    #[serde(default = "OptionalENConfig::default_main_node_rate_limit_window_ms")]
+    main_node_rate_limit_window_ms: u64,
```

Comment:
A large window such as 1s could lead to stampeding requests, which we'd probably want to avoid. It's possible to hard-code a small window (IMO, it makes sense to do on the EN level, not in the `L2Client` definition, to allow for more flexibility) and re-compute the number of requests in this window based on the desired RPS. E.g., if a window is 50ms and the EN is configured with 100 rps limit, this would lead to 5 requests per window. Does this sound good?

Reply:
@slowli I think this is the best - it would not only make sure we don't do more than X requests per second, but distribute them (almost) uniformly over time. 

Re:  config vs no-config - probably indeed doing this the way you suggest and still configuring target RPS PER SECOND is a better interface.